From 4587cdbc14a76647aeb81789cb2708601ffd9bef Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Fri, 5 Oct 2018 09:16:27 +0530
Subject: [PATCH 0440/1851] v4l: xilinx-gamma: Correct default values when max
 width, height dt property absent

commit 3477c876101ff17039ee77b4bcbbafb892d73800 from
https://github.com/Xilinx/linux-xlnx.git

Revert to 3840x2160 as default value of max width and height to
maintain compatibility instead of 8192x4320.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-gamma.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-gamma.c b/drivers/media/platform/xilinx/xilinx-gamma.c
index 667c48a4fd86..d113fca9b624 100644
--- a/drivers/media/platform/xilinx/xilinx-gamma.c
+++ b/drivers/media/platform/xilinx/xilinx-gamma.c
@@ -28,6 +28,8 @@
 #define XGAMMA_MIN_WIDTH	(64)
 #define XGAMMA_MAX_WIDTH	(8192)
 #define XGAMMA_DEF_WIDTH	(1280)
+#define XGAMMA_DEF_MAX_HEIGHT	(2160)
+#define XGAMMA_DEF_MAX_WIDTH	(3840)
 
 #define XGAMMA_AP_CTRL			(0x0000)
 #define XGAMMA_GIE			(0x0004)
@@ -358,7 +360,7 @@ static int xg_parse_of(struct xgamma_dev *xg)
 
 	rval = of_property_read_u32(node, "xlnx,max-height", &xg->max_height);
 	if (rval < 0) {
-		xg->max_height = XGAMMA_MAX_HEIGHT;
+		xg->max_height = XGAMMA_DEF_MAX_HEIGHT;
 	} else if (xg->max_height > XGAMMA_MAX_HEIGHT ||
 		   xg->max_height < XGAMMA_MIN_HEIGHT) {
 		dev_err(dev, "Invalid height in dt");
@@ -367,7 +369,7 @@ static int xg_parse_of(struct xgamma_dev *xg)
 
 	rval = of_property_read_u32(node, "xlnx,max-width", &xg->max_width);
 	if (rval < 0) {
-		xg->max_width = XGAMMA_MAX_WIDTH;
+		xg->max_width = XGAMMA_DEF_MAX_WIDTH;
 	} else if (xg->max_width > XGAMMA_MAX_WIDTH ||
 		   xg->max_width < XGAMMA_MIN_WIDTH) {
 		dev_err(dev, "Invalid width in dt");
-- 
2.31.1

